### PR TITLE
task 4

### PR DESCRIPTION
**Pull request by sharksystems**
Added test cases C384, C401, C405, C390, C393, C593, C594, C595, C593, C594, C595

### CODE REVIEW

File: tests/unit_creation.spec.ts [hunk 1]
```diff
@@ -180,4 +180,137 @@ test.describe('Unit creation tests', async () => {
         await unitCreationPage.verifySelectedTabIsHighlighted(1);
     });
 
+    test('C384 - Verify uploading the same image', async ({ unitCreationPage }) => {
+        (await unitCreationPage.getTabTitleByNumber(1)).click();
```
Comment (AndreyCar) [inline tests/unit_creation.spec.ts]: Remove these brackets

File: tests/unit_creation.spec.ts [hunk 2]
```diff
@@ -180,4 +180,137 @@ test.describe('Unit creation tests', async () => {
...
+
+        await unitCreationPage.uploadImagesToBlock(1, ["valid_jpeg.jpeg"]);
+        await unitCreationPage.assertErrorPopupVisible();
+        await unitCreationPage.clickOutsidePopup();
+        await unitCreationPage.assertErrorPopupNotVisible();
+        await unitCreationPage.verifyImageInBlock(1, '');
+    });
+
+    test('C401 - Verify uploading invalid file type', async ({ unitCreationPage }) => {
+        (await unitCreationPage.getTabTitleByNumber(1)).click();
```
Comment (AndreyCar) [inline tests/unit_creation.spec.ts]: Remove these brackets

File: tests/unit_creation.spec.ts [hunk 3]
```diff
@@ -180,4 +180,137 @@ test.describe('Unit creation tests', async () => {
...
+
+        await unitCreationPage.uploadImagesToBlock(0, ["webp_image.webp"]);
+        await unitCreationPage.assertErrorPopupVisible();
+        await unitCreationPage.clickOutsidePopup();
+        await unitCreationPage.assertErrorPopupNotVisible();
+        await unitCreationPage.verifyImageInBlock(0, '');
+    });
+
+    test('C405 - Verify uploading image of size more than 20MB', async ({ unitCreationPage }) => {
+        (await unitCreationPage.getTabTitleByNumber(1)).click();
```
Comment (AndreyCar) [inline tests/unit_creation.spec.ts]: Remove these brackets

File: tests/unit_creation.spec.ts [hunk 4]
```diff
@@ -180,4 +180,137 @@ test.describe('Unit creation tests', async () => {
...
+
+        await unitCreationPage.uploadImagesToBlock(0, ["image_over_20mb.png"]);
+        await unitCreationPage.assertErrorPopupVisible();
+        await unitCreationPage.clickOutsidePopup();
+        await unitCreationPage.assertErrorPopupNotVisible();
+        await unitCreationPage.verifyImageInBlock(0, '');
+    });
+
+    test('C390 - Verify "Назад" button (2. Photo Tab)', async ({ unitCreationPage }) => {
+        (await unitCreationPage.getTabTitleByNumber(1)).click();
```
Comment (AndreyCar) [inline tests/unit_creation.spec.ts]: Remove these brackets

File: tests/unit_creation.spec.ts [hunk 5]
```diff
@@ -180,4 +180,137 @@ test.describe('Unit creation tests', async () => {
...
+        await unitCreationPage.assertCategorySelectionTitleText();
+        await unitCreationPage.assertListingTitleInputTitleText();
+        await unitCreationPage.assertManufacturerInputElements();
+        await unitCreationPage.assertModelInputTitleText();
+        await unitCreationPage.assertDescriptionInputTitle();
+        await unitCreationPage.assertMapSelectionTitle();
+    });
+
+    test('C393 - Verify "Далі" button (2. Photo Tab)', async ({ unitCreationPage }) => {
+        (await unitCreationPage.getTabTitleByNumber(1)).click();
```
Comment (AndreyCar) [inline tests/unit_creation.spec.ts]: Remove these brackets

File: tests/unit_creation.spec.ts [hunk 6]
```diff
@@ -180,4 +180,137 @@ test.describe('Unit creation tests', async () => {
...
+
+        await unitCreationPage.uploadImagesToBlock(1, ["valid_jpeg2.jpeg", "valid_jpeg3.jpeg", "valid_jpg.jpg", "valid_jpg2.jpg", "valid_jpg3.jpg", "valid_jpg4.jpg", "valid_png.png", "valid_png2.png", "valid_png3.png", "valid_png4.png", "valid_png5.png"]);
+        await unitCreationPage.assertTotalNumberOfImageBlocks(12);
+        for(let i = 0; i < 12; i++) {
+            await unitCreationPage.getImageBlockSource(i);
+        }
+    });
+
+    test('C594 - Verify image drag & drop', async ({ unitCreationPage }) => {
+        (await unitCreationPage.getTabTitleByNumber(1)).click();
```
Comment (AndreyCar) [inline tests/unit_creation.spec.ts]: Remove these brackets

File: tests/unit_creation.spec.ts [hunk 7]
```diff
@@ -180,4 +180,137 @@ test.describe('Unit creation tests', async () => {
...
+        let firstBlockImage = await unitCreationPage.getImageBlockSource(0);
+        let secondBlockImage = await unitCreationPage.getImageBlockSource(1);
+
+        await unitCreationPage.dragAndDropImage(1, 0);
+        await unitCreationPage.verifyImageInBlock(0, secondBlockImage);
+        await unitCreationPage.verifyImageInBlock(1, firstBlockImage);
+    });
+
+    test('C595 - Verify image deleting', async ({ unitCreationPage }) => {
+        (await unitCreationPage.getTabTitleByNumber(1)).click();
```
Comment (AndreyCar) [inline tests/unit_creation.spec.ts]: Remove these brackets

**Review by AndreyCar — CHANGES_REQUESTED** [review]

File: tests/unit_creation.spec.ts (hunk 1 above)
Comment (sharksystems) [reply tests/unit_creation.spec.ts]: getTabTitleByNumber and similar methods didnt really need to be asynchronous, fixed in the commit

**Review by AndreyCar — APPROVED** [review]

**Pull request merged**